src/modules/hip/kernel/non_silent_region_detection.hpp

```diff
@@ -0,0 +1,410 @@
+#include <hip/hip_runtime.h>
+#include "rpp_hip_common.hpp"
+
+// -------------------- Set 0 -  moving mean square kernel device helpers --------------------
+
+// calculate the position in shared memory to avoid bank conflicts
+__host__ __device__ __forceinline__ int smem_pos(int pos)
+{
+    return pos + (pos >> 5); // since shared memory banks considered is 32
+}
+
+__device__ __forceinline__ void compute_prefix_sum(float *input, uint bufferLength)
+{
+    int offset = 1;
+    int offsetA = offset - 1;
+    int offsetB = 2 * offset - 1;
+    int tid = hipThreadIdx_x;
```

Comment:
You can directly do int twiceThreadID = 2 * hipThreadIdx_x; since tid is not really used, you seem to only use 2 * tid

Reply:
And on L32, you can use if (hipThreadIdx_x == 0) directly